src/util.jl

```diff
@@ -0,0 +1,17 @@
+"""
+    compute_frames!(actions::Vector{AA}) where AA <: AbstractAction
+
+Set action.frames.frames to the computed frames for each action in actions.
+"""
+function compute_frames!(actions::Vector{AA}) where AA <: AbstractAction
+    last_frames = nothing
+    for action in actions
+        if last_frames === nothing && get_frames(action) === nothing
+            throw(ArgumentError("Frames need to be defined explicitly, at least for the first frame."))
```

Comment:
Could you make this ArgumentError a little more precise? I think what you could say is something like "Frames must be set explicitly in either the initial `Action` or `BackgroundAction` object." Wording is up to you but I find "at least for the first time" a little too vague.

Reply:
How about: "Frames need to be defined explicitly in the inital AbstractAction like Action/BackgroundAction or SubAction."?

Reply:
Sounds good to me!